doc: build files: move the list to new page (#13154)
nrfconnect/sdk-nrf: doc/nrf/config_and_build/output_build_files.rst

.. _app_build_output_files:

Output build files (image files)
################################

The building process produces each time an :term:`image file`.

The image file can refer to an *executable*, a *program*, or an *ELF file*.
As one of the last build steps, the linker processes all object files by locating code, data, and symbols in sections in the final ELF file.
The linker replaces all symbol references to code and data with addresses.

> A symbol table is created which maps addresses to symbol names, which is used by debuggers.

Reviewer: ```suggestion
A symbol table that maps addresses to symbol names is created.
This table is used by debuggers.
```

> When an ELF file is converted into another format, such as HEX or binary, the symbol table is lost.

Reviewer: ```suggestion
When an ELF file is converted to another format, such as HEX or binary, the symbol table is lost.
```

Depending on the application and the SoC, you can use one or several images.

The |NCS| build system places output images in the :file:`<build folder>/zephyr` folder.

.. _app_build_output_files_common:

Common output build files
*************************

The following table lists build files that can be generated as output when building firmware for supported :ref:`build targets <app_boards>`.

> The table includes files for single-core and multi-core programming scenarios for both |VSC| and command line building methods.

Reviewer: ```suggestion
The table includes files for single-core and multi-core programming scenarios for both |VSC| and command-line building methods.
```

Which files you are going to use depends on the application configuration and not directly on the type of SoC you are using.
The following scenarios are possible:

* Single-image - Only one firmware image file is generated for a single core.
* Multi-image - Two or more firmware image files are generated for a single core.

>   You can read more about this scenario in :ref:`ug_multi_image`.

Reviewer: ```suggestion
  For more details about this scenario, see :ref:`ug_multi_image`.
```

|                                          | The *<file_name>* includes manufacturer's code, image type, file version, and comment                 |                                                                                                           |
|                                          | (customizable by user, sample name by default).                                                       |                                                                                                           |
|                                          | For example: :file:`127F-0141-01020003-light_switch.zigbee`.                                          |                                                                                                           |
+------------------------------------------+-------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------+

.. _app_build_mcuboot_output:

MCUboot output build files
**************************

> .. note::
>     MCUboot's :doc:`direct-xip mode <mcuboot:design>` and the related :kconfig:option:`CONFIG_BOOT_BUILD_DIRECT_XIP_VARIANT` Kconfig option are currently supported only on the single-core devices such as the nRF52 Series (:ref:`more information <ug_nrf52_developing_ble_fota_mcuboot_direct_xip_mode>`).

Reviewer: ```suggestion
.. note::
   MCUboot's :doc:`direct-xip mode <mcuboot:design>` and the related :kconfig:option:`CONFIG_BOOT_BUILD_DIRECT_XIP_VARIANT` Kconfig option are currently supported only on single-core devices, such as nRF52 Series devices.
   For more details, see the :ref:`ug_nrf52_developing_ble_fota_mcuboot_direct_xip_mode` section of the :ref:`ug_nrf52_developing` page.
```
